coral-spark/src/main/java/com/linkedin/coral/spark/SparkSqlRewriter.java

```diff
@@ -65,6 +64,23 @@ public SqlNode visit(SqlCall call) {
     return super.visit(call);
   }
 
+  /**
+   * Check if the SqlDataTypeSpec contains `ROW`
+   */
+  private boolean containsSqlRowTypeSpec(SqlDataTypeSpec sqlDataTypeSpec) {
+    if (sqlDataTypeSpec instanceof SqlRowTypeSpec) {
+      return true;
+    }
+    if (sqlDataTypeSpec instanceof SqlArrayTypeSpec) {
+      return containsSqlRowTypeSpec(((SqlArrayTypeSpec) sqlDataTypeSpec).getElementTypeSpec());
+    }
+    if (sqlDataTypeSpec instanceof SqlMapTypeSpec) {
+      return containsSqlRowTypeSpec(((SqlMapTypeSpec) sqlDataTypeSpec).getKeyTypeSpec())
+          || containsSqlRowTypeSpec(((SqlMapTypeSpec) sqlDataTypeSpec).getValueTypeSpec());
```

Comment:
is there a UT to point out why we needed to expand the criteria for containsSqlRowTypeSpec()? 
Can there be a sqlCall - CAST(operand to Array<row>) ?